packages/calypso-e2e/src/lib/components/comments-likes-component.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Internal dependencies
+ */
+import { BaseContainer } from '../base-container';
+
+/**
+ * Type dependencies
+ */
+import { Page, ElementHandle } from 'playwright';
+
+const selectors = {
+	topLevelComments: '.comment-list .depth-1:not(.children)',
+	likeButton: '.comment-like-link',
+	notLiked: '.comment-not-liked',
+	liked: '.comment-liked',
+};
+
+/**
+ * Component representing the like component of a comment.
+ *
+ * @augments {BaseContainer}
+ */
+export class CommentsLikesComponent extends BaseContainer {
```

Comment:
Would it make sense to merge it with the `CommentsComponent`?

Reply:
I can look into that for this PR. Do you think it would make sense to also rename the `PostLikesComponent` into a `PostPage`, analogous to the change proposed for comments?

Reply:
I have made changes using an approach that I hope is not too controversial: inner/nested class. 

Previously (as you have noticed) there were multiple components/pages at play:

- PostLikesComponent to interact with the post like button.
- CommentComponent to interact with the comment field.
- CommentsLikesComponent to interact with the comment like button.
- PublishedPostsListPage to represent the page on the published site that lists all posts.

It occurred to me that there was no page object to represent a single published post. In my mind, a revised hierarchy should look like this:

```
          PublishedPostsListPage
                   v
          PublishedPostsPage
        /                                 \
PostLikesComponent       CommentsComponent

```

I used inner classes as I felt that `PostLikes` and `Comments` deserve some separation from the parent class `PublishedPost` to make it easier to conceptualize.

I'd like to hear what you think. Do you think it may be anti-pattern to use inner classes?